wallet-new/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -1001,6 +1078,139 @@ instance BuildableSafeGen NewAddress where
+-- m /      44' /      1815' /       0' /      0 /             1
+--
+-- NOTE See:
+--   - https://github.com/satoshilabs/slips/blob/master/slip-0044.md
+--   - https://github.com/satoshilabs/slips/pull/123
+data AddressPath = AddressPath
+  { addrpathPurpose      :: AddressLevel
+  , addrpathCoinType     :: AddressLevel
+  , addrpathAccount      :: AddressLevel
+  , addrpathChange       :: AddressLevel
+  , addrpathAddressIndex :: AddressLevel
+  } deriving (Show, Eq, Generic)
+
+deriveJSON Serokell.defaultOptions ''AddressPath
+
+makeLensesFor
+  [ ("addrpathAccount", "account")
+  , ("addrpathChange", "change")
+  , ("addrpathAddressIndex", "addressIndex")
+  ] ''AddressPath
+
+purpose :: Getter AddressPath AddressLevel
+purpose = to addrpathPurpose
+
+coinType :: Getter AddressPath AddressLevel
+coinType = to addrpathCoinType
+
+instance ToSchema AddressPath where
+    declareNamedSchema =
+        genericSchemaDroppingPrefix "addr" (\(--^) props -> props
+            & ("purpose"      --^ "44, refers to BIP-44")
+            & ("coinType"     --^ "1815 for ADA (Ada Lovelace's birthdate)")
+            & ("account"      --^ "Account index, used as child index in BIP-32 derivation")
+            & ("change"       --^ "0 if external (e.g. payment addr), 1 if internal (e.g. change addr)")
+            & ("addressIndex" --^ "Address index counter, incremental")
+        )
+
+instance Arbitrary AddressPath where
+    arbitrary = AddressPath <$> arbitrary
+                            <*> arbitrary
+                            <*> arbitrary
+                            <*> arbitrary
+                            <*> arbitrary
+
+deriveSafeBuildable ''AddressPath
+instance BuildableSafeGen AddressPath where
+    buildSafeGen sl AddressPath{..} = bprint (""
+        %"{ purpose="%buildSafe sl
+        %", coin_type="%buildSafe sl
+        %", account="%buildSafe sl
+        %", change="%buildSafe sl
+        %", address_index="%buildSafe sl
+        %" }")
+        addrpathPurpose
+        addrpathCoinType
+        addrpathAccount
+        addrpathChange
+        addrpathAddressIndex
+
+-- Smart constructor to create BIP44 derivation paths
+--
+-- NOTE Our account indexes are already referred to as "hardened" and
+-- are therefore referred to as indexes above 2^31 == maxBound `div` 2 + 1
+-- AddressPath makes it explicit whether a path should be hardened or not
+-- instead of relying on a convention on number.
+mkAddressPathBIP44
+    :: IsChangeAddress -- ^ Whether this is an internal (for change) or external address (for payments)
+    -> Account         -- ^ Underlying account
+    -> Either Text AddressPath
+mkAddressPathBIP44 (IsChangeAddress isChange) acct = AddressPath
+    <$> pure (AddressLevelHardened 44)
+    <*> pure (AddressLevelHardened 1815)
+    <*> pure (word32ToAddressLevel $ accIndex acct)
+    <*> pure (AddressLevelNormal $ if isChange then 1 else 0)
+    <*> checkWord31 AddressLevelNormal (getAddressIndex acct)
```

Comment:
Could also be `AddressPath (AddressLevelHardened 44) (AddressLevelHardened 1815) (word32ToAddressLevel $ accIndex acct) ... <$> checkWord31 AddressLevelNormal (getAddressIndex acct)`